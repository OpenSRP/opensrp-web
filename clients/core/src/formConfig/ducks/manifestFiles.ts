import {
    fetchActionCreatorFactory,
    removeActionCreatorFactory,
    reducerFactory,
    getItemsByIdFactory,
    getItemsArrayFactory,
    getItemByIdFactory,
} from '../../store/ducks/baseDux';

/** manifest files interface */
export interface ManifestFilesTypes {
    created_at: string;
    id: string;
    identifier: string;
    is_draft: boolean;
    is_json_validator: boolean;
    jursdiction: string;
    label: string;
    module: string;
    version: string;
}

/** reducer name */
export const reducerName = 'manifestFiles';

/** manifest files Reducer */
const reducer = reducerFactory<ManifestFilesTypes>(reducerName);

// action
/** fetch manifest files to store action */
export const fetchManifestFiles = fetchActionCreatorFactory<ManifestFilesTypes>(reducerName, 'id');

/** clear manifest files data in store action*/
export const removeManifestFiles = removeActionCreatorFactory(reducerName);

// selectors
export const getAllManifestFilesById = getItemsByIdFactory<ManifestFilesTypes>(reducerName);
export const getManifestFilesById = getItemByIdFactory<ManifestFilesTypes>(reducerName);
export const getAllManifestFilesArray = getItemsArrayFactory<ManifestFilesTypes>(reducerName);

export default reducer;
