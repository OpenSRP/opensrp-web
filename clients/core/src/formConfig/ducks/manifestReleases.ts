import {
    fetchActionCreatorFactory,
    removeActionCreatorFactory,
    reducerFactory,
    getItemsByIdFactory,
    getItemsArrayFactory,
    getItemByIdFactory,
    getTotalRecordsFactory,
} from '../../store/ducks/baseDux';

/** json field interface */
export interface ManifestJsonFieldType {
    forms_version: string;
    identifiers: string[];
}

/** manifest releses interface */
export interface ManifestReleasesTypes {
    appId: string;
    appVersion: string;
    createdAt: string;
    identifier: string;
    json: string;
    updatedAt: string;
}

/** reducer name */
export const reducerName = 'manifestReleases';

/** manifest releases Reducer */
const reducer = reducerFactory<ManifestReleasesTypes>(reducerName);

// action
/** fetch manifest releases to store action */
export const fetchManifestReleases = fetchActionCreatorFactory<ManifestReleasesTypes>(reducerName, 'identifier');

/** clear manifest releases data in store action*/
export const removeManifestReleases = removeActionCreatorFactory(reducerName);

// selectors
export const getAllManifestReleasesById = getItemsByIdFactory<ManifestReleasesTypes>(reducerName);
export const getManifestReleasesById = getItemByIdFactory<ManifestReleasesTypes>(reducerName);
export const getAllManifestReleasesArray = getItemsArrayFactory<ManifestReleasesTypes>(reducerName);
export const getTotalManifestReleasesRecords = getTotalRecordsFactory(reducerName);

export default reducer;
