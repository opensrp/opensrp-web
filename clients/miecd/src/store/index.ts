import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
    router: connectReducer as any /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener(reducers => {
    store.replaceReducer(combine(reducers));
});

export default store;
