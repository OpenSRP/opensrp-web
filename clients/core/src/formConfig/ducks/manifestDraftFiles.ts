import {
    fetchActionCreatorFactory,
    removeActionCreatorFactory,
    reducerFactory,
    getItemsByIdFactory,
    getItemsArrayFactory,
    getItemByIdFactory,
} from '@opensrp/reducer-factory';
import { ManifestFilesTypes } from './manifestFiles';

/** reducer name */
export const reducerName = 'manifestDraftFiles';

/** manifest draft files Reducer */
const reducer = reducerFactory<ManifestFilesTypes>(reducerName);

// action
/** fetch manifest draft files to store action */
export const fetchManifestDraftFiles = fetchActionCreatorFactory<ManifestFilesTypes>(reducerName, 'id');

/** clear manifest draft files data in store action*/
export const removeManifestDraftFiles = removeActionCreatorFactory(reducerName);

// selectors
export const getAllManifestDraftFilesById = getItemsByIdFactory<ManifestFilesTypes>(reducerName);
export const getManifestDraftFilesById = getItemByIdFactory<ManifestFilesTypes>(reducerName);
export const getAllManifestDraftFilesArray = getItemsArrayFactory<ManifestFilesTypes>(reducerName);

export default reducer;
