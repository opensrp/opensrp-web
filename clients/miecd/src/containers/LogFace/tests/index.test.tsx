import { history } from '@onaio/connected-reducer-registry';
import reducerRegistry from '@onaio/redux-reducer-registry';
import { ConnectedRouter } from 'connected-react-router';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Provider } from 'react-redux';
import ConnectedLogFace from '..';
import { DEFAULT_NUMBER_OF_LOGFACE_ROWS, PREGNANCY } from '../../../constants';
import store from '../../../store';
import reducer, { fetchLocations, fetchUserId, fetchUserLocations, reducerName } from '../../../store/ducks/locations';
import { fetchSms, removeSms } from '../../../store/ducks/sms_events';
import { communes, districts, provinces, villages } from '../../HierarchichalDataTable/test/fixtures';
import { smsSlice } from './fixtures';
import { userLocations } from './userLocationFixtures';

reducerRegistry.register(reducerName, reducer);

describe('containers/LogFace', () => {
    const props = { module: PREGNANCY };
    afterEach(() => {
        store.dispatch(removeSms);
    });
    beforeEach(() => {
        jest.resetAllMocks();
        store.dispatch(fetchLocations(provinces));
        store.dispatch(fetchLocations(districts));
        store.dispatch(fetchLocations(communes));
        store.dispatch(fetchLocations(villages));
        store.dispatch(fetchUserLocations(userLocations));
        store.dispatch(fetchUserId('515ad0e9-fccd-4cab-8861-0ef3ecb831e0'));
    });
    it('renders without crashing', () => {
        shallow(
            <Provider store={store}>
                <ConnectedLogFace {...props} />
            </Provider>,
        );
    });

    it('renders correctly', () => {
        store.dispatch(fetchSms(smsSlice));
        const wrapper = mount(
            <Provider store={store}>
                <ConnectedRouter history={history}>
                    <ConnectedLogFace {...props} />
                </ConnectedRouter>
            </Provider>,
        );

        wrapper.update();
        expect(toJson(wrapper.find('table'))).toMatchSnapshot('table snapshot');
        expect(toJson(wrapper.find('.location-type-filter'))).toMatchSnapshot('filter div');
        expect(toJson(wrapper.find('input#input'))).toMatchSnapshot('search div');
        expect(toJson(wrapper.find('#logface_title'))).toMatchSnapshot('logface title');
        expect(toJson(wrapper.find('.paginator'))).toMatchSnapshot('paginator');
        wrapper.unmount();
    });

    it('it renders only 10 items per page ', () => {
        store.dispatch(fetchSms(smsSlice));
        const wrapper = mount(
            <Provider store={store}>
                <ConnectedRouter history={history}>
                    <ConnectedLogFace {...props} />
                </ConnectedRouter>
            </Provider>,
        );

        // + 1 is added here to unclude the header `tr`
        expect(wrapper.find('tr').length).toBe(DEFAULT_NUMBER_OF_LOGFACE_ROWS + 1);
        wrapper.unmount();
    });

    it('search works correctly', () => {
        store.dispatch(fetchSms(smsSlice));
        const wrapper = mount(
            <Provider store={store}>
                <ConnectedRouter history={history}>
                    <ConnectedLogFace {...props} />
                </ConnectedRouter>
            </Provider>,
        );

        expect(wrapper.find('input').length).toBe(1);
        // wrapper.find('input').simulate('change', { target: { value: '1569837448461' } });
        // + 1 is added here to unclude the header `tr`
        expect(wrapper.find('tr').length).toBe(DEFAULT_NUMBER_OF_LOGFACE_ROWS + 1);
    });
});
