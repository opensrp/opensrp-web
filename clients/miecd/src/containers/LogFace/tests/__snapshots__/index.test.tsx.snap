// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/LogFace renders correctly: filter div 1`] = `
Array [
  <div
    className="location-type-filter"
  >
    <span>
      Risk Level
    </span>
    <Dropdown
      a11y={true}
      active={false}
      addonType={false}
      direction="down"
      inNavbar={false}
      isOpen={false}
      nav={false}
      setActiveFromChild={false}
      toggle={[Function]}
    >
      <Manager>
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <DropdownToggle
            aria-haspopup={true}
            caret={true}
            color="secondary"
            disabled={false}
            id="dropdown-basic"
            variant="success"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <Button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle"
                  color="secondary"
                  disabled={false}
                  id="dropdown-basic"
                  innerRef={[Function]}
                  onClick={[Function]}
                  tag="button"
                  variant="success"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-label={null}
                    className="dropdown-toggle btn btn-secondary"
                    disabled={false}
                    id="dropdown-basic"
                    onClick={[Function]}
                    type="button"
                    variant="success"
                  >
                    <span>
                      Select risk
                    </span>
                  </button>
                </Button>
              </InnerReference>
            </Reference>
          </DropdownToggle>
          <DropdownMenu
            flip={true}
            tag="div"
          >
            <div
              aria-hidden={true}
              className="dropdown-menu"
              role="menu"
              tabIndex="-1"
            >
              <DropdownItem
                key="red"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  red
                </button>
              </DropdownItem>
              <DropdownItem
                key="high"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  high
                </button>
              </DropdownItem>
              <DropdownItem
                key="low"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  low
                </button>
              </DropdownItem>
              <DropdownItem
                key="no risk"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  no risk
                </button>
              </DropdownItem>
              <DropdownItem
                key="all"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  all
                </button>
              </DropdownItem>
            </div>
          </DropdownMenu>
        </div>
      </Manager>
    </Dropdown>
  </div>,
  <div
    className="location-type-filter"
  >
    <span>
      Select Location
    </span>
    <Dropdown
      a11y={true}
      active={false}
      addonType={false}
      direction="down"
      inNavbar={false}
      isOpen={false}
      nav={false}
      setActiveFromChild={false}
      toggle={[Function]}
    >
      <Manager>
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <DropdownToggle
            aria-haspopup={true}
            caret={true}
            color="secondary"
            disabled={false}
            id="dropdown-basic"
            variant="success"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <Button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle"
                  color="secondary"
                  disabled={false}
                  id="dropdown-basic"
                  innerRef={[Function]}
                  onClick={[Function]}
                  tag="button"
                  variant="success"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-label={null}
                    className="dropdown-toggle btn btn-secondary"
                    disabled={false}
                    id="dropdown-basic"
                    onClick={[Function]}
                    type="button"
                    variant="success"
                  >
                    <span>
                      Select Location
                    </span>
                  </button>
                </Button>
              </InnerReference>
            </Reference>
          </DropdownToggle>
          <DropdownMenu
            flip={true}
            tag="div"
          >
            <div
              aria-hidden={true}
              className="dropdown-menu"
              role="menu"
              tabIndex="-1"
            >
              <DropdownItem
                key="Yen Phu Village"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Yen Phu Village
                </button>
              </DropdownItem>
              <DropdownItem
                key="all"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  all
                </button>
              </DropdownItem>
            </div>
          </DropdownMenu>
        </div>
      </Manager>
    </Dropdown>
  </div>,
  <div
    className="location-type-filter"
  >
    <span>
      Type
    </span>
    <Dropdown
      a11y={true}
      active={false}
      addonType={false}
      direction="down"
      inNavbar={false}
      isOpen={false}
      nav={false}
      setActiveFromChild={false}
      toggle={[Function]}
    >
      <Manager>
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <DropdownToggle
            aria-haspopup={true}
            caret={true}
            color="secondary"
            disabled={false}
            id="dropdown-basic"
            variant="success"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <Button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle"
                  color="secondary"
                  disabled={false}
                  id="dropdown-basic"
                  innerRef={[Function]}
                  onClick={[Function]}
                  tag="button"
                  variant="success"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-label={null}
                    className="dropdown-toggle btn btn-secondary"
                    disabled={false}
                    id="dropdown-basic"
                    onClick={[Function]}
                    type="button"
                    variant="success"
                  >
                    <span>
                      Select Type
                    </span>
                  </button>
                </Button>
              </InnerReference>
            </Reference>
          </DropdownToggle>
          <DropdownMenu
            flip={true}
            tag="div"
          >
            <div
              aria-hidden={true}
              className="dropdown-menu"
              role="menu"
              tabIndex="-1"
            >
              <DropdownItem
                key="Response Report"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Response Report
                </button>
              </DropdownItem>
              <DropdownItem
                key="Red Alert Report"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Red Alert Report
                </button>
              </DropdownItem>
              <DropdownItem
                key="Social Determinants"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Social Determinants
                </button>
              </DropdownItem>
              <DropdownItem
                key="ANC Visit"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  ANC Visit
                </button>
              </DropdownItem>
              <DropdownItem
                key="Delivery Planning"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Delivery Planning
                </button>
              </DropdownItem>
              <DropdownItem
                key="Pregnancy Detection"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Pregnancy Detection
                </button>
              </DropdownItem>
              <DropdownItem
                key="Pregnancy Identification"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Pregnancy Identification
                </button>
              </DropdownItem>
              <DropdownItem
                key="Pregnancy registration"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Pregnancy registration
                </button>
              </DropdownItem>
              <DropdownItem
                key="ANC Report"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  ANC Report
                </button>
              </DropdownItem>
              <DropdownItem
                key="Home Visit Report"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Home Visit Report
                </button>
              </DropdownItem>
              <DropdownItem
                key="Birth Report"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Birth Report
                </button>
              </DropdownItem>
              <DropdownItem
                key="Death Report"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Death Report
                </button>
              </DropdownItem>
              <DropdownItem
                key="Postnatal and Newborn Care"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Postnatal and Newborn Care
                </button>
              </DropdownItem>
              <DropdownItem
                key="Nutrition Registration"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Nutrition Registration
                </button>
              </DropdownItem>
              <DropdownItem
                key="Nutrition Report"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Nutrition Report
                </button>
              </DropdownItem>
              <DropdownItem
                key="Monthly Nutrition Report"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Monthly Nutrition Report
                </button>
              </DropdownItem>
              <DropdownItem
                key="Departure Code"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Departure Code
                </button>
              </DropdownItem>
              <DropdownItem
                key="Refusal Code"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Refusal Code
                </button>
              </DropdownItem>
              <DropdownItem
                key="Account Check"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Account Check
                </button>
              </DropdownItem>
              <DropdownItem
                key="all"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  all
                </button>
              </DropdownItem>
            </div>
          </DropdownMenu>
        </div>
      </Manager>
    </Dropdown>
  </div>,
]
`;

exports[`containers/LogFace renders correctly: logface title 1`] = `
<h2
  id="logface_title"
>
  Log Face - Pregnancy
</h2>
`;

exports[`containers/LogFace renders correctly: paginator 1`] = `
<div
  className="paginator"
>
  <Paginator
    ariaLabel="page navigation"
    endLabel="last"
    nextLabel="next"
    onPageChange={[Function]}
    pageLimit={5}
    pageNeighbours={3}
    previousLabel="previous"
    startLabel="first"
    totalRecords={32}
  >
    <BasePaginator
      ariaLabel="page navigation"
      currentPage={1}
      endLabel="last"
      nextLabel="next"
      onPageChange={[Function]}
      pageChangeHandler={[Function]}
      pageLimit={5}
      pageNeighbours={3}
      previousLabel="previous"
      startLabel="first"
      totalRecords={32}
    >
      <Pagination
        aria-label="page navigation"
        listTag="ul"
        size="sm"
        tag="nav"
      >
        <nav
          aria-label="page navigation"
          className=""
        >
          <ul
            className="pagination pagination-sm"
          >
            <PaginationItem
              className="page-item disabled"
              tag="li"
            >
              <li
                className="page-item disabled page-item"
              >
                <PaginationLink
                  aria-label="first"
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                  tag="a"
                >
                  <a
                    aria-label="first"
                    className="page-link page-link"
                    href="#"
                    onClick={[Function]}
                  >
                    <span
                      aria-hidden="true"
                    >
                      first
                    </span>
                    <span
                      className="sr-only"
                    >
                      first
                    </span>
                  </a>
                </PaginationLink>
              </li>
            </PaginationItem>
            <PaginationItem
              className="page-item disabled"
              tag="li"
            >
              <li
                className="page-item disabled page-item"
              >
                <PaginationLink
                  aria-label="previous"
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                  tag="a"
                >
                  <a
                    aria-label="previous"
                    className="page-link page-link"
                    href="#"
                    onClick={[Function]}
                  >
                    <span
                      aria-hidden="true"
                    >
                      previous
                    </span>
                    <span
                      className="sr-only"
                    >
                      previous
                    </span>
                  </a>
                </PaginationLink>
              </li>
            </PaginationItem>
            <PaginationItem
              className="page-item  active"
              key="0"
              tag="li"
            >
              <li
                className="page-item  active page-item"
              >
                <PaginationLink
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                  tag="a"
                >
                  <a
                    className="page-link page-link"
                    href="#"
                    onClick={[Function]}
                  >
                    1
                  </a>
                </PaginationLink>
              </li>
            </PaginationItem>
            <PaginationItem
              className="page-item "
              key="1"
              tag="li"
            >
              <li
                className="page-item  page-item"
              >
                <PaginationLink
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                  tag="a"
                >
                  <a
                    className="page-link page-link"
                    href="#"
                    onClick={[Function]}
                  >
                    2
                  </a>
                </PaginationLink>
              </li>
            </PaginationItem>
            <PaginationItem
              className="page-item "
              key="2"
              tag="li"
            >
              <li
                className="page-item  page-item"
              >
                <PaginationLink
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                  tag="a"
                >
                  <a
                    className="page-link page-link"
                    href="#"
                    onClick={[Function]}
                  >
                    3
                  </a>
                </PaginationLink>
              </li>
            </PaginationItem>
            <PaginationItem
              className="page-item "
              key="3"
              tag="li"
            >
              <li
                className="page-item  page-item"
              >
                <PaginationLink
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                  tag="a"
                >
                  <a
                    className="page-link page-link"
                    href="#"
                    onClick={[Function]}
                  >
                    4
                  </a>
                </PaginationLink>
              </li>
            </PaginationItem>
            <PaginationItem
              className="page-item  "
              tag="li"
            >
              <li
                className="page-item   page-item"
              >
                <PaginationLink
                  aria-label="next"
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                  tag="a"
                >
                  <a
                    aria-label="next"
                    className="page-link page-link"
                    href="#"
                    onClick={[Function]}
                  >
                    <span
                      aria-hidden="true"
                    >
                      next
                    </span>
                    <span
                      className="sr-only"
                    >
                      next
                    </span>
                  </a>
                </PaginationLink>
              </li>
            </PaginationItem>
            <PaginationItem
              className="page-item "
              tag="li"
            >
              <li
                className="page-item  page-item"
              >
                <PaginationLink
                  aria-label="last"
                  className="page-link "
                  href="#"
                  onClick={[Function]}
                  tag="a"
                >
                  <a
                    aria-label="last"
                    className="page-link  page-link"
                    href="#"
                    onClick={[Function]}
                  >
                    <span
                      aria-hidden="true"
                    >
                      last
                    </span>
                    <span
                      className="sr-only"
                    >
                      last
                    </span>
                  </a>
                </PaginationLink>
              </li>
            </PaginationItem>
          </ul>
        </nav>
      </Pagination>
    </BasePaginator>
  </Paginator>
</div>
`;

exports[`containers/LogFace renders correctly: search div 1`] = `
<input
  className="form-control logface-search"
  disabled={false}
  id="input"
  name="input"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Search ID, Reporter, Patients"
  type="text"
/>
`;

exports[`containers/LogFace renders correctly: table snapshot 1`] = `
<table
  className="table table-borderless table-striped"
>
  <thead
    id="header"
  >
    <tr>
      <th
        className="default-width"
      >
        Event Date
      </th>
      <th
        className="default-width"
      >
        Location
      </th>
      <th
        className="default-width"
      >
        SMS Type
      </th>
      <th
        className="default-width"
      >
        Reporter
      </th>
      <th
        className="default-width"
      >
        Patient
      </th>
      <th
        className="small-width"
      >
        Age
      </th>
      <th
        className="large-width"
      >
        Message
      </th>
      <th
        className="default-width"
      >
        Risk Level
      </th>
    </tr>
  </thead>
  <tbody
    id="body"
  >
    <tr
      key="1570173602178"
    >
      <td
        className="default-width"
      >
        2019-10-04
      </td>
      <td
        className="default-width"
      >
        Yen Phu Village
      </td>
      <td
        className="default-width"
      >
        Response Report
      </td>
      <td
        className="default-width"
      >
        Husna Hariz
      </td>
      <td
        className="default-width"
      >
        <Link
          to="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
        >
          <LinkAnchor
            href="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
            navigate={[Function]}
          >
            <a
              href="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
              onClick={[Function]}
            >
              100P9K-20191001-01
            </a>
          </LinkAnchor>
        </Link>
      </td>
      <td
        className="small-width"
      >
        3d
      </td>
      <td
        className="large-width"
      >
        Symptoms = No symptoms  
Intervention = Ambulance On Time 
Health Status = 
      </td>
      <td
        className="default-width"
      >
        <Link
          to="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
        >
          <LinkAnchor
            href="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
            navigate={[Function]}
          >
            <a
              href="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
              onClick={[Function]}
            >
              <riskColoring
                risk="no risk"
              >
                <span
                  className="badge badge-success"
                  id="default-width"
                >
                  <div>
                    <p>
                      No Risk
                    </p>
                  </div>
                </span>
              </riskColoring>
            </a>
          </LinkAnchor>
        </Link>
      </td>
    </tr>
    <tr
      key="1570173399115"
    >
      <td
        className="default-width"
      >
        2019-10-04
      </td>
      <td
        className="default-width"
      >
        Yen Phu Village
      </td>
      <td
        className="default-width"
      >
        Red Alert Report
      </td>
      <td
        className="default-width"
      >
        Husna Hariz
      </td>
      <td
        className="default-width"
      >
        <Link
          to="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
        >
          <LinkAnchor
            href="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
            navigate={[Function]}
          >
            <a
              href="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
              onClick={[Function]}
            >
              100P9K-20191001-01
            </a>
          </LinkAnchor>
        </Link>
      </td>
      <td
        className="small-width"
      >
        3d
      </td>
      <td
        className="large-width"
      >
        Symptoms = Coughing, Fever 
Location = At home 
Interventions = 
      </td>
      <td
        className="default-width"
      >
        <Link
          to="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
        >
          <LinkAnchor
            href="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
            navigate={[Function]}
          >
            <a
              href="/pregnancy_log_face/child_patient_detail/100P9K-20191001-01"
              onClick={[Function]}
            >
              <riskColoring
                risk="red"
              >
                <span
                  className="badge badge-danger"
                  id="default-width"
                >
                  <div>
                    <p>
                      Red Alert
                    </p>
                  </div>
                </span>
              </riskColoring>
            </a>
          </LinkAnchor>
        </Link>
      </td>
    </tr>
    <tr
      key="1570171129630"
    >
      <td
        className="default-width"
      >
        2019-10-04
      </td>
      <td
        className="default-width"
      >
        Yen Phu Village
      </td>
      <td
        className="default-width"
      >
        Pregnancy Registration
      </td>
      <td
        className="default-width"
      >
        HMN
      </td>
      <td
        className="default-width"
      >
        <Link
          to="/pregnancy_log_face/patient_detail/100PAH"
        >
          <LinkAnchor
            href="/pregnancy_log_face/patient_detail/100PAH"
            navigate={[Function]}
          >
            <a
              href="/pregnancy_log_face/patient_detail/100PAH"
              onClick={[Function]}
            >
              100PAH
            </a>
          </LinkAnchor>
        </Link>
      </td>
      <td
        className="small-width"
      >
        30y 9m
      </td>
      <td
        className="large-width"
      >
        EDD = 09-10-2019
Gravidity = 3
Parity = 1
Weight = 55 
Height = 145
BMI = 26
Previous Gynaecology History = Previous pregnancy too close or too far, Home Delivery
Current Symptoms = Heavy sweating, Coughing
      </td>
      <td
        className="default-width"
      >
        <Link
          to="/pregnancy_log_face/patient_detail/100PAH"
        >
          <LinkAnchor
            href="/pregnancy_log_face/patient_detail/100PAH"
            navigate={[Function]}
          >
            <a
              href="/pregnancy_log_face/patient_detail/100PAH"
              onClick={[Function]}
            >
              <riskColoring
                risk="red"
              >
                <span
                  className="badge badge-danger"
                  id="default-width"
                >
                  <div>
                    <p>
                      Red Alert
                    </p>
                  </div>
                </span>
              </riskColoring>
            </a>
          </LinkAnchor>
        </Link>
      </td>
    </tr>
  </tbody>
</table>
`;
